src/data/list.js

```diff
@@ -0,0 +1,176 @@
+            lists._id = lists._id.toString();
+            returnList.push(lists);
+        }
+
+        return returnList;
+    },
+
+    /**
+    * Gets a list with a given id.
+    * @param {String} listId The lists's id.
+    * @param {String} boardId The board's id.
+    * @returns A list object, if list does not exist, throws error.
+    */
+    getListById: async (listId, boardId) => {
+        if(!listId || !error_handler.checkObjectId(listId)) 
+            throw new Error("listId is not valid.");
+
+        if(!boardId || !error_handler.checkObjectId(boardId)) 
+            throw new Error("boardId is not valid.");
+        
+        const board = await boardData.readById(boardId);
+
+        let returnList;
+        for(let y=0; y<board.lists.length; y++) {
+            let list = board.lists[y];
+            if(list._id.toString() === listId) {
+                returnList = list;
+            }
+        }
+        
+        if(!returnList) throw new Error('List not found');
+
+        return returnList;
+    },
+
+    /**
+    * Creates a new list.
+    * @param {string} listName The name of the new list.
+    * @param {String} boardId The id of the board the list is added to.
+    * @returns A board object, otherwise throws error if list wasn't added.
+    */
+    addList: async (listName, boardId) => {
+        if(!boardId || !error_handler.checkObjectId(boardId)) 
+            throw new Error("id is not valid.");
+
+        if(!listName || !error_handler.checkNonEmptyString(listName))
+            throw new Error("listName is not valid.")
+
+        const boardCollection = await boards();
+
+        let listId = new ObjectId();
+        const newList = {
+            _id: listId,
+            listName: listName.trim(),
+            cardIds: []
+        };
+
+        const updateInfo = await boardCollection.updateOne(
+            { _id: ObjectId(boardId) },
+            { $push: { lists: newList } }
+        );
+
+        if(!updateInfo.matchedCount && !updateInfo.modifiedCount) throw new Error('Update failed');
+        return boardData.readById(boardId);
+    },
+
+    /**
+    * Adds card to end of list.
+    * @param {string} listId The id of the list the card will be added to.
+    * @param {string} cardId The id of the added card.
+    * @param {string} boardId The id of the board the list is in.
+    * @returns True if successfully added, otherwise throws Error.
+    */
+    addCardIdtoList: async (listId, cardId, boardId) => {
+        if(!boardId || !error_handler.checkObjectId(boardId)) 
+            throw new Error("boardId is not valid.");
+
+        if(!listId || !error_handler.checkObjectId(listId)) 
+            throw new Error("listId is not valid.");
+        
+        if(!cardId || !error_handler.checkObjectId(cardId)) 
+            throw new Error("cardId is not valid.");
+
+        let list = this.getListById(listId, boardId);
+
+        let cardArray = list.cardIds;
+        cardArray.push(cardId);
```

Comment:
I'd prefer just using $addToSet: in the update to add a cardId without allowing duplicates to be more robust. I believe this isn't bad, though.